cpu/sensors.py

```diff
@@ -0,0 +1,116 @@
+#! /usr/bin/env python
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2016 IBM
+# Author: Harish <harisrir@linux.vnet.ibm.com>
+#
+
+"""
+Test for sensors command
+"""
+import platform
+
+from avocado import Test
+from avocado import main
+from avocado.utils import process
+from avocado.utils import distro
+from avocado.utils.software_manager import SoftwareManager
+
+# TODO: Add possible errors of sensors command
+ERRORS = ['I/O error']
+
+
+class Sensors(Test):
+
+    """
+    Run sensors command on ppc architectures.
+    """
+
+    @staticmethod
+    def check_errors(cmd):
+        """
+        Checks any errors in command result
+        """
+        errs = []
+        s_output = process.system_output(cmd)
+        for error in ERRORS:
+            if error in s_output:
+                errs.append(error)
+        return errs
+
+    def test(self):
+        """
+        Test for sensors command
+        """
+        s_mg = SoftwareManager()
+        d_distro = distro.detect()
+        if d_distro.name == "Ubuntu":
+            if not s_mg.check_installed("lm-sensors") and not s_mg.install(
+                    "lm-sensors"):
+                self.error('Need sensors to run the test')
+        else:
+            if not s_mg.check_installed("lm_sensors") and not s_mg.install(
+                    "lm_sensors"):
+                self.error('Need sensors to run the test')
```

Comment:
The repetition of code can be a sign to have a function.  It could receive the package list, install those not already  installed and returns true/false on success/error.
But my concern here is the lack of distros. For example, this test will fail on debian if `lm-sensors` is not present.
Also, this check is everywhere on our tests in this repo. Maybe we should make it part of SoftwareManager. I will open a Trello card for Avocado so we can evaluate that.